### PR TITLE
fix(Tree): Switcher loading icon and checkbox should be vertical align when @tree-title-height is modified

### LINKED ISSUE #15955: 通过主题定制修改@tree-title-height的值会导致tree-switcher-loading-icon的位置不再居中
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://github.com/ant-design/ant-design/blob/a26397843c97b17d16a33e91de0b58898ff980fe/components/tree/style/index.less#L71](https://github.com/ant-design/ant-design/blob/a26397843c97b17d16a33e91de0b58898ff980fe/components/tree/style/index.less#L71)

### Steps to reproduce
查看上述代码发现，@{tree-prefix-cls}-switcher-loading-icon 的 width 和 height 为固定的 24px，因此当 @tree-title-height 为默认的 24px 时一切正常，但是调整 @tree-title-height 为 36px 之后，就会发现 tree-switcher-loading-icon 的位置不再是垂直居中，而是偏向左上方。

### What is expected?
希望能在调整 @tree-title-height 的值之后，tree-switcher-loading-icon 依旧保持局中位置。

### What is actually happening?
调整 @tree-title-height 的值为 36px 之后，tree-switcher-loading-icon 的位置不再是垂直居中，而是偏向左上方。

| Environment | Info |
|---|---|
| antd | 3.16.2 |
| React | 16.7.0 |
| System | Mac OS X 10.14 |
| Browser | Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:60.0) Gecko/20100101 Firefox/60.0 |

---
因为可以通过查看源代码直接发现问题，就不再另外提供重现链接了。

### PR DESCRIPTION
**Pull request by dengfuping**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

- Close #15955

### 📝 Changelog

- English Changelog:

- 🐞 Fix switcher loading icon and checkbox not vertical align for Tree when @tree-title-height is modified. #15955

- Chinese Changelog (optional):

- 🐞 修复定制 @tree-title-height 变量时 Tree 组件中的 `loading` 图标和 `checkbox` 没有垂直居中的问题。#15955

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
屏幕快照 2019-04-09 上午11 36 46

**Comment by dengfuping** [comment]
- Fixed checkbox not vertical align.

File: components/tree/style/index.less [hunk 1]
```diff
@@ -120,13 +120,17 @@
     }
     span {
       &.@{tree-prefix-cls}-checkbox {
-        margin: 4px 4px 0 2px;
+        position: initial;
+        display: inline-flex;
```
Comment (zombieJ) [inline components/tree/style/index.less]: 可以换种方式不？

Comment (dengfuping) [reply components/tree/style/index.less]: @zombieJ  Done ~，改为设置上下 padding 来实现垂直居中了

File: components/tree/style/index.less [hunk 2]
```diff
@@ -120,13 +120,16 @@
     }
     span {
       &.@{tree-prefix-cls}-checkbox {
-        margin: 4px 4px 0 2px;
+        top: initial;
+        height: @tree-title-height;
+        margin: 0 4px 0 2px;
+        padding: calc((@tree-title-height - 16px) / 2) 0;
```
Comment (afc163) [inline components/tree/style/index.less]: 这个 calc 貌似没什么意义。

Comment (dengfuping) [reply components/tree/style/index.less]: - calc 在 less 下会被忽略单位全部按照百分比计算，已改为:
```css
padding: ~'calc((@{tree-title-height} - 16px) / 2)' 0;
```

Comment (afc163) [reply components/tree/style/index.less]: 不加 calc 也是一样的，less 会自动计算。

Comment (dengfuping) [reply components/tree/style/index.less]: 已去掉 `calc`

**Review by zombieJ — APPROVED** [review]

**Pull request merged**